cpptod.dd

```diff
@@ -913,6 +915,130 @@ SecondVariant.method();
 ------
 
 
+$(HR)
+
+$(H3 $(LNAME2 functioninputparameters, Function Input Parameters))
+
+$(H4 The C++ Way)
+
+        $(B C++ does not support explicit input parameters.) If you use a
+        $(I pass-by-value) parameter $(D void function(int inputOnlyImplicitly)) it can just
+        be an input parameter implicitly. If you use a $(I pass-by-reference) parameter
+        usually $(D const) is applied to the parameter
+        $(D void foo(const int& inputOnlyParameterImplicitly)) that a forbidden
+        modification of the parameter in the function leads to a compilation error.
+
+$(CPPCODE
+void foo(int implicitInputParameterPassedByValue)
+{
+    // some code dependent on implicitInputParameterPassedByValue
+    ...
+}
+
+void foo(int& implicitInputParameterPassedByReference)
+{
+    // some code dependent on implicitInputParameterPassedByReference
+    ...
+}
+)
+
+The functions are used as follows:
+
+$(CPPCODE
+int inputArgumentPassedByValue = 42;
+foo(inputArgumentPassedByValue);
+// inputArgumentPassedByValue = 42
+
+int inputArgumentPassedByReference = 42;
+foo(inputArgumentPassedByReference);
+// inputArgumentPassedByReference = 42
+)
+
+$(H4 The D Way)
+
+        D supports explicit input parameters.
+
+------
+void foo(in int inputParameter)
+{
+    // some code dependent on inputParameter
+    ...
+}
+------
+
+The function is used as follows:
+
+$(RUNNABLE_EXAMPLE
+---------
+void foo(in int inputParameter)
+{
+    ++inputParameter; // Compiler will throw an error: cannot modify const expression inputParameter
+}
+
+void main()
+{
+    int inputArgument = 42;
+    foo(inputArgument);
+    assert(inputArgument == 42);
+}
+---------
+)
+
+$(HR)
+
+$(H3 $(LNAME2 functionoutputparameters, Function Output Parameters))
+
+$(H4 The C++ Way)
+
+        $(B C++ does not support explicit output parameters.) You can use a reference
+        parameter to allow the function's parameter to serve as input and output
+        to/from the function. Input only parameters $(I called-by-reference) not
+        $(I called-by-value) are usually declared as such by applying $(D const) to the
+        parameter $(D void foo(const int& inputParameter)). Then the remaining uncertainty
+        is wether the parameter is an output only or an input and output parameter.
+
+$(CPPCODE
+void foo(int& inputAndOutputParameter)
+{
+    inputAndOutputParameter = 42;
+}
+)
+
+The function is used as follows ($(I call-by-reference)):
```

Comment:
This sounds weird. Ideas:
- This function is called as follows:
- Calling by reference is done as follows